File: Java/4.21.java
/*
Given a line of text as input, output the number of characters excluding spaces, periods, or commas.

Ex: If the input is:

Listen, Mr. Jones, calm down.
the output is:

21
Note: Account for all characters that aren't spaces, periods, or commas (Ex: "r", "2", "!").
*/

import java.util.Scanner;

public class LabProgram {
   public static void main(String[] args) {
      Scanner scan = new Scanner(System.in);
      String userText;
      int count = 0;
   
      userText = scan.nextLine();
      
      for (int i = 0; i < userText.length(); i++) {
         count++;
         if ((userText.charAt(i) == ' ') || (userText.charAt(i) == '.') || (userText.charAt(i) == ',')) {
            count--;
         }
      }
      System.out.printf("%d\n", count);
   }
}

File: CPP/4.21.cpp
/*
Given a line of text as input, output the number of characters excluding spaces, periods, exclamation points, or commas.

Ex: If the input is:

Listen, Mr. Jones, calm down.
the output is:

21
*/

#include <iostream>
#include <string>
using namespace std;

int main() {
   string userText;
   // Add more variables as needed
   
   getline(cin, userText);  // Gets entire line, including spaces. 

   /* Type your code here. */
   int count = 0;
   
   for (unsigned int i = 0; i < userText.length(); i++) {
      if ((userText[i] != ' ') && (userText[i] != '.') && (userText[i] != '!') && (userText[i] != ',')) {
         count++;
      }
   }
   
   printf("%d\n", count);

   return 0;
}
